config/config.go

```diff
@@ -1,47 +1,51 @@
 package config
 
 import (
-	"gopkg.in/yaml.v2"
 	"io/ioutil"
 	"path/filepath"
 	"regexp"
+
+	"gopkg.in/yaml.v2"
 )
 
-// RawConfig - used to unmarshall the raw config file
-type RawConfig struct {
-	S3 rawResourceType `yaml:"s3"`
+// Config - the config object we pass around
+type Config struct {
+	S3       ResourceType `yaml:"s3"`
+	IAMUsers ResourceType `yaml:"IAMUsers"`
 }
 
-type rawResourceType struct {
-	IncludeRule rawFilterRule `yaml:"include"`
-	ExcludeRule rawFilterRule `yaml:"exclude"`
+type ResourceType struct {
+	IncludeRule FilterRule `yaml:"include"`
+	ExcludeRule FilterRule `yaml:"exclude"`
 }
 
-type rawFilterRule struct {
-	NamesRE []string `yaml:"names_regex"`
+type FilterRule struct {
+	NamesRE []Expression `yaml:"names_regex"`
 }
 
-// Config - the config object we pass around
-// that is a parsed version of RawConfig
-type Config struct {
-	S3 ResourceType
+type Expression struct {
+	RE regexp.Regexp
 }
 
-// ResourceType - the include and exclude
-// rules for a resource type
-type ResourceType struct {
-	IncludeRule FilterRule
-	ExcludeRule FilterRule
-}
+// UnmarshalText - Internally used by yaml.Unmarshal to unmarshall an Expression field
+func (expression *Expression) UnmarshalText(data []byte) error {
+	var pattern string
 
-// FilterRule - contains regular expressions or plain text patterns
-// used to match against a resource type's properties
-type FilterRule struct {
-	NamesRE []*regexp.Regexp
```

Comment:
NIT: could we also use a more descriptive name here: `NamesRegExp` would do fine too - I just found myself wondering later on what this was.

Reply:
Yeap, naming is hard :disappointed: 
Do you have any more suggestions? `NamesRegExp` is fine for me too, but I don't know if it is clear/explicit enough :thinking:

Reply:
Agreed - finding the right name is difficult. I am happy to proceed with whatever makes sense to you most!